src/azure-cli/azure/cli/command_modules/synapse/operations/accesscontrol.py

```diff
@@ -0,0 +1,142 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.util import CLIError
+from azure.synapse.accesscontrol.models import RoleAssignmentOptions
+from azure.cli.core.util import is_guid
+from azure.graphrbac.models import GraphErrorException
+from msrestazure.azure_exceptions import CloudError
+from .._client_factory import cf_synapse_client_accesscontrol_factory, cf_graph_client_factory
+
+
+# List Synapse Role Assignment
+def list_role_assignments(cmd, workspace_name, role=None, assignee=None):
+    # get role id
+    role_id = _resolve_role_id(cmd, role, workspace_name)
+    # get object_id
+    object_id = _resolve_object_id(cmd, assignee, fallback_to_object_id=True)
+
+    client = cf_synapse_client_accesscontrol_factory(cmd.cli_ctx, workspace_name)
+    role_assignments = client.get_role_assignments(role_id, object_id)
+
+    # TODO:
+    # Currently, when only `ObjectId` is specified, the cmdlet returns incorrect result.
+    # Filter from client side as a workaround
+    if object_id:
+        role_assignments = [x for x in role_assignments if x.principal_id == object_id]
+    return role_assignments
+
+
+# Show Synapse Role Assignment By Id
+def get_role_assignment_by_id(cmd, workspace_name, role_assignment_id):
+    client = cf_synapse_client_accesscontrol_factory(cmd.cli_ctx, workspace_name)
+    return client.get_role_assignment_by_id(role_assignment_id)
+
+
+# Delete Synapse Role Assignment
+def delete_role_assignment_by_id(cmd, workspace_name, ids=None, assignee=None, role=None, delete_all=None):
```

Comment:
Maybe "delete_role_assignment" is a better name as you may delete multiple assignments.